### PR TITLE
API modification: support combined dipole training mode and global dipole test mode.

### PR DESCRIPTION
**Pull request by YWolfeee**
dp train / freeze / test check has finished, but tensorboard used in loss/tensor.py has not been tested yet.

Author: Haotian Ye (YWolfeee)
Date: 9, Apr, 2021.

### CODE REVIEW

File: deepmd/entrypoints/main.py [hunk 1]
```diff
@@ -232,10 +232,10 @@ def parse_args(args: Optional[List[str]] = None):
     )
     parser_tst.add_argument(
         "-a",
-        "--atomic-energy",
+        "--atomic",
```
Comment (njzjz) [inline deepmd/entrypoints/main.py:235]: Please fix unit tests if you import a breaking change.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: deepmd/loss/tensor.py [hunk 2]
```diff
@@ -12,24 +12,60 @@ class TensorLoss () :
...
             self.scale = 1.0
+
+        # YHT: added for global / local dipole combination
+        if self.atomic is True: # upper regulation, will control the lower behavior
+            self.local_weight,self.global_weight = 1.0,0.0
+        elif self.atomic is False: # upper regulation, will control the lower behavior
+            self.local_weight,self.global_weight = 0.0,1.0
+        else: # self.atomic is None, let the loss parameter decide which mode to use
+            if jdata is not None:
+                self.local_weight = jdata.get('atomic_' + self.tensor_name,None)
```
Comment (amcadmus) [inline deepmd/loss/tensor.py:34]: It seems that the name of the prefactors are `atomic_dipole` and `dipole`.   

I would suggest to change them to `pref_atomic_dipole` and `pref_dipole`.

File: deepmd/loss/tensor.py [hunk 3]
```diff
@@ -12,24 +12,60 @@ class TensorLoss () :
...
+                    assert self.local_weight > 0.0, AssertionError('assign a zero weight to local dipole without setting a global weight')
+                    self.local_weight,self.global_weight = 1.0,0.0
+                    self.atomic = True
+                else:   # Both are not None
+                    self.atomic = True if self.local_weight != 0.0 else False
+                    assert (self.local_weight >0.0) or (self.global_weight>0.0), AssertionError('can not assian zero weight to both local and global mode')
+                    # normalize
+                    temp_sum = self.local_weight + self.global_weight
+                    self.local_weight   /=  temp_sum
+                    self.global_weight  /=  temp_sum
```
Comment (amcadmus) [inline deepmd/loss/tensor.py:57]: I suggest not doing this normalization. Leave the prefactors the same as those provided by the user.

File: deepmd/entrypoints/test.py [hunk 4]
```diff
@@ -563,19 +564,31 @@ def test_dipole(
...
+    if has_atom_dipole == False:
+        dipole = np.reshape(dipole,(dipole.shape[0], -1, 3))
+        atoms = dipole.shape[1]
+        dipole = np.sum(dipole,axis=1)
+
     l2f = l2err(dipole - test_data["dipole"][:numb_test])
 
+    if has_atom_dipole == False:
+        l2f = l2f / atoms
+
```
Comment (amcadmus) [inline deepmd/entrypoints/test.py:591]: The design here is not good.

We assume that `has_atom_dipole == False`, then by line 576, the label is ensured to be global. The atoms on line 584 is always 1.

When `has_atom_dipole == False`, the user wants to test the error in global dipole. The RMSE should be normalized by number of atoms. This information can be get from `test_data`.

**Review by amcadmus — CHANGES_REQUESTED** [review]

File: deepmd/loss/tensor.py (hunk 2 above)
Comment (YWolfeee) [reply deepmd/loss/tensor.py:34]: I will fix this

File: deepmd/loss/tensor.py (hunk 3 above)
Comment (YWolfeee) [reply deepmd/loss/tensor.py:57]: I will fix this.

File: deepmd/entrypoints/test.py (hunk 4 above)
Comment (YWolfeee) [reply deepmd/entrypoints/test.py:591]: I don't think it's true. Suppose has_atom_dipole is false.
The frozen model will output local dipole. Therefore, it should be shaped as (nframs x sel_atom x 3).
However, the label is (nframes x 3) in this case.
Therefore, it's necessary to store natoms and do normalization

Comment (amcadmus) [reply deepmd/entrypoints/test.py:591]: That's clear to me

**Comment by YWolfeee** [comment]
Finish the above problem, including ut test and some advise from Han Wang. Tensorboard hasn't tested yet.

Author: Haotian Ye (YWolfeee)
Date: 12, Apr, 2021.

**Review by amcadmus — APPROVED** [review]

**Pull request merged**